### DISCUSSION TITLE
N320 or N321 Series Config Files

### DISCUSSION THREAD

**Discussion by rolandmartinson7890** [srsran/srsRAN_Project#226] [General Help]
Hello srsRAN users. Has anyone ever run on the N321 or N320 SDRs. I have B210s and they connect well without a problem but the N321 (with the configuration below) doesnt allow the UE to see the network:

amf:
  addr: 127.0.1.5                                             # The address or hostname of the AMF.
  bind_addr: 127.0.0.1                                          # A local IP that the gNB binds to for traffic from the AMF.
  
ru_sdr:
  device_driver: uhd                # The RF driver name.
  device_args: addr=192.168.40.9,clock_source=internal,send_frame_size=8000,recv_frame_size=8000, master_clock_rate=245.76e6            # Optionally pass arguments to the selected RF driver.
  clock: internal                      # Specify the clock source used by the RF.
  sync:                               # Specify the sync source used by the RF.
  srate: 61.44 #61.44 #122.88                      # RF sample rate might need to be adjusted according to selected bandwidth.
  tx_gain: 55                       # Transmit gain of the RF might need to adjusted to the given situation.
  rx_gain: 40                       # Receive gain of the RF might need to adjusted to the given situation.
  lo_offset: 25                     #should be half the size of bandwidth
  expert_cfg: 
    low_phy_thread_profile: quad

cell_cfg:
  dl_arfcn: 628140                                              # ARFCN of the downlink carrier (center frequency).
  band: 78                                                      # The NR band.
  channel_bandwidth_MHz: 20 #100                                     # Bandwith in MHz. Number of PRBs will be automatically derived. 20MHz for 23.04
  common_scs: 30                                                # Subcarrier spacing in kHz used for data.
  plmn: "00101"                                                 # PLMN broadcasted by the gNB.
  tac: 1                                                        # Tracking area code (needs to match the core configuration).
  pci: 1                                                        # Physical cell ID.
  nof_antennas_dl: 2                                            #For configuring MIMO 2x2 - B210 only supports 2x2
  nof_antennas_ul: 2                                            #For configuring MIMO 2x2
  ssb:
    ssb_period: 10
  pdsch:
    mcs_table: qam256
log:
  filename: /tmp/iq.log
  all_level: info
  phy_rx_symbols_filename: /tmp/iq.log

pcap:
  mac_enable: false                                             # Set to true to enable MAC-layer PCAPs.
  mac_filename: /tmp/gnb_mac.pcap                               # Path where the MAC PCAP is stored.
  ngap_enable: false                                            # Set to true to enable NGAP PCAPs.
  ngap_filename: /tmp/gnb_ngap.pcap                             # Path where the NGAP PCAP is stored.

**Comment by pgawlowicz** [comment]
Could you add the following to the UE config and see whether it works with x310?

```
[expert]
lte_sample_rates = true
```

**Reply by rolandmartinson7890** [reply]
I am using the N321. I will however try that out when I get access to the devices again and let you know.

**Reply by pgawlowicz** [reply]
Hi @rolandmartinson7890, any progress with this issue?

**Reply by rolandmartinson7890** [reply]
I have been able to connect using N321 a while ago. I did not use this (lte_sample_rates = true).

It was however a one time instance and my B210s seem to have a much stable connection in comparison to the more powerful and expensive SDRs like N321 and X310.

I reverted back to the B210s for the rest of my experiments because I didnt wanna waste time configuring.

**Reply by pgawlowicz** [reply]
Thanks for your feedback!
